cmd/registry/cmd/label.go

```diff
@@ -0,0 +1,260 @@
+	"github.com/apigee/registry/gapic"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/names"
+	"github.com/spf13/cobra"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+var labelFilter string
+var labelOverwrite bool
+var labelsToSet map[string]string
+var labelsToClear []string
+
+func init() {
+	rootCmd.AddCommand(labelCmd)
+	labelCmd.Flags().StringVar(&labelFilter, "filter", "", "Filter selected resources")
+	labelCmd.Flags().BoolVar(&labelOverwrite, "overwrite", false, "Overwrite existing labels")
+}
+
+var labelCmd = &cobra.Command{
+	Use:   "label RESOURCE KEY_1=VAL_1 ... KEY_N=VAL_N",
+	Short: "Label resources in the API Registry",
+	Args:  cobra.MinimumNArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.TODO()
+		client, err := connection.NewClient(ctx)
+		if err != nil {
+			log.Fatalf("%s", err.Error())
+		}
+
+		taskQueue := make(chan core.Task, 1024)
+		workerCount := 64
+		for i := 0; i < workerCount; i++ {
+			core.WaitGroup().Add(1)
+			go core.Worker(ctx, taskQueue)
+		}
+
+		labelsToClear = make([]string, 0)
+		labelsToSet = make(map[string]string)
+		for _, labeling := range args[1:] {
+			if len(labeling) > 1 && strings.HasSuffix(labeling, "-") {
+				labelsToClear = append(labelsToClear, labeling[0:len(labeling)-1])
+			} else {
+				pair := strings.Split(labeling, "=")
+				if len(pair) != 2 {
+					log.Fatalf("%q must have the form \"key=value\" (value can be empty) or \"key-\" (to remove the key)", labeling)
+				}
+				if pair[0] == "" {
+					log.Fatalf("%q is invalid because it specifies an empty key", labeling)
+				}
+				labelsToSet[pair[0]] = pair[1]
+			}
+		}
+
+		err = matchAndHandleLabelCmd(ctx, client, taskQueue, args[0])
+		if err != nil {
+			log.Fatalf("%s", err.Error())
+		}
+
+		close(taskQueue)
+		core.WaitGroup().Wait()
+	},
+}
+
+type labelApiTask struct {
+	ctx    context.Context
+	client connection.Client
+	api    *rpc.Api
+}
+
+type labelVersionTask struct {
+	ctx     context.Context
+	client  connection.Client
+	version *rpc.ApiVersion
+}
+
+type labelSpecTask struct {
+	ctx    context.Context
+	client connection.Client
+	spec   *rpc.ApiSpec
+}
+
+func (task *labelApiTask) Name() string {
+	return "label " + task.api.Name
+}
+
+func (task *labelApiTask) Run() error {
+	if task.api.Labels == nil {
+		task.api.Labels = make(map[string]string)
+	}
+	if !labelOverwrite {
+		for k, _ := range labelsToSet {
+			if v, ok := task.api.Labels[k]; ok {
+				return fmt.Errorf("%q already has a value (%s), and --overwrite is false", k, v)
+			}
+		}
+	}
+	for _, k := range labelsToClear {
+		delete(task.api.Labels, k)
+	}
+	if len(labelsToSet) > 0 {
+		for k, v := range labelsToSet {
+			task.api.Labels[k] = v
+		}
+	}
+	req := &rpc.UpdateApiRequest{
+		Api: task.api,
+		UpdateMask: &field_mask.FieldMask{
+			Paths: []string{"labels"},
+		},
+	}
+	_, err := task.client.UpdateApi(task.ctx, req)
+	return err
+}
+
+func (task *labelVersionTask) Name() string {
```

Comment:
Good point - and I think this has come up before. We should act on this in a separate PR because this is determined by the [Tasks interface](https://github.com/apigee/registry/blob/92b14e9ac410da3d87b1b1e2a3da5c77e121e7c0/cmd/registry/core/tasks.go#L26). #170